grammars/silver/compiler/extension/tuple/README.md

```diff
@@ -0,0 +1,54 @@
+# Tuple Syntax
+
+Tuples use a comma "`,`" as an infix operator to seperate individual tuple elements. They are implemented inductively using the construction of ordered pairs, e.g. we forward a tuple `(a, b, c, d)` to `Pair<a Pair<b Pair<c d>>>`. This means the following examples are equivalent:
+
+    ("I", ("am", "a", (5, "tuple"))) = ("I", "am", "a", 5, "tuple")
```

Comment:
Not sure if this information should go in the README or on the website. I want to give a brief overview of how tuples work but I'm not sure if this is brief enough here. In this section I could just point to the website to explain usage?

Reply:
I'm not sure how I feel about having README.md files scattered in the code base - this would be the only non-`.sv` file under `grammars/`, as far as I know.  The info before the "Implementation" section here (don't to mention patterns too!) could fit on the website under http://melt.cs.umn.edu/silver/ref/ext/.

The rest of the implementation description can probably move into the source code as doc comments in appropriate places - so this would (eventually) end up on the website as well if we decide to put up generated docs for the compiler internals.  

The way that this is implemented doesn't really have `,` as an infix operator, by the way - if it were so you could write just `[String, Integer]` without the parentheses.  But it can be helpful to think about that way, I guess.

Reply:
I think the benefit of having `README.md` files in extension directories is to provide documentation on that extension for Silver developers, so if anyone needs to make changes to the extension they have a quick introduction to it.

With that interpretation, the part before `Implementation` doesn't really belong here, since that is more for users of the extension than for developers of it.

Reply:
> I think the benefit of having README.md files in extension directories is to provide documentation on that extension for Silver developers, so if anyone needs to make changes to the extension they have a quick introduction to it.

Yes, but that is what doc comments are for.  There are a number of comments in the Silver compiler source that should become generated doc comments once that is working.  My main issue with using `README.md` files is that these won't be included in the generated documentation.

Reply:
I can translate the `README.md` into doc comments but I'm not entirely sure if there is a consensus on this so I'm not sure how I should proceed.

Reply:
If @ericvanwyk thinks that we should have README files in extension grammars then I guess this is fine.  If these were doc comments instead they would still be in the source tree and also get added to the website.  But that can always be fixed later if/when we start putting generated docs from the `silver:compiler` grammars on our website.

Reply:
Every repo should have a README since this might be some reader's first entrance into that material.  It might be that the best thing to have in the README are not much more than pointers to the Silver web page for how to use, in this case, the extension, and to the generated doc-comments that describe the implementation of the specifications in the repo.  In the case of this one I think it makes sense to say what the purpose of the extension is, a little bit about what it provides and then pointers to the relevant web pages that have the documentation. We do want to avoid multiple copies of large bits of documentation - but a paragraph or two that lives here in the README and also somewhere doesn't seem unreasonable. At least until we have the Silver documentation sorted out. What is here now might be more than we need - but again - we need to have the other documentation sorted out before we judge what is here as being too much or too little.